File: code/DVH_2022_1.py
#!.venv/bin/python3
# %%

"""
author: Daan Van Hauwermeiren
AoC 2022 day 1
"""
import sys

def main() -> tuple[int, int]:
    fn = sys.argv[1]

    calories = [0,]
    with open(fn, mode="r") as f:
        raw = [i.strip() for i in f.readlines()]

    idx = 0
    for line in raw:
        if len(line) == 0:
            idx += 1
            calories.append(0)
            continue
        else:
            calories[idx] += int(line)

    calories_top3 = sorted(calories)[-3:]

    solution_1 = calories[0]
    solution_2 = sum(calories_top3)

    return solution_1, solution_2


if __name__ == '__main__':
    solutions = main()
    print(solutions)
# %%


# %%

File: code/DVH_2022_1.jl
#=
author: Daan Van Hauwermeiren
AoC 2022 day 1
=#
using Pkg
Pkg.activate(".", io=devnull)
using Pipe

# run via 
# julia ./code/DVH_2022_1.jl ./data/input_2022_1.txt

function main(args)
    fn = args[1]
    
    raw = read(fn, String);
    # NOTE: these two lines might be merged for a nice one-liner.
    inventory_raw = @pipe rstrip(raw) |> split(_, "\n\n")
    calories_summed = @pipe split.(inventory_raw, "\n") .|> parse.(Int, _) .|> sum |> sort
    
    solution_1 = calories_summed[end]
    solution_2 = sum(calories_summed[end-2:end])
    @show solution_1, solution_2
end
main(ARGS)

